moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp

```diff
@@ -63,7 +63,6 @@ moveit::core::RobotState* ompl_interface::TSStateStorage::getStateStorage() cons
   if (it == thread_states_.end())
   {
     st = new moveit::core::RobotState(start_state_);
-    st->setToDefaultValues();  // avoid uninitialized memory
```

Comment:
@henningkayser sorry for being late to the party :( I added this line last summer while debugging nasty memory issues. (probably this one #2239). Valgrind reported uninitialized memory here so I changed it. Looking at the `robot_state.cpp` code now I cannot see how this could have caused an issue.

My best guess is that I never removed it (or really understood it) because of my limited experience at the time and it was not catched in a review because CI did not complain.

Reply:
@JeroenDM thanks for responding anyway! ;) While removing this line fixes the linked issues, I think that we are still dealing with some memory issues. So maybe you were onto something there. At least we've seen flaky constrained planner tests and some uninitialized memory during the tests as well. We also haven't looked into this too much yet. Good that this is clarified, though.